cmd/endpoint.go

```diff
@@ -349,12 +349,22 @@ func CreateEndpoints(serverAddr string, args ...string) (string, EndpointList, S
 					err = fmt.Errorf("path '%s' can not be served by different port on same address", endpoint.Path)
 					return serverAddr, endpoints, setupType, err
 				}
-
 				pathIPMap[endpoint.Path] = IPSet.Union(hostIPSet)
 			} else {
 				pathIPMap[endpoint.Path] = hostIPSet
 			}
 		}
+
+		// Check if all IPs in the path IP map are local, we do not allow more than
+		// one IPs on same server to serve same disk paths.
+		for _, endpoint := range endpoints {
+			if IPSet, ok := pathIPMap[endpoint.Path]; ok {
+				if len(localIP4.Intersection(IPSet).ToSlice()) > 1 {
```

Comment:
1. You could move this check in https://github.com/minio/minio/blob/master/cmd/endpoint.go#L352 to avoid one more looping.
2. You would need to use like https://github.com/minio/minio/blob/master/cmd/net.go#L296

Reply:
> You could move this check in https://github.com/minio/minio/blob/master/cmd/endpoint.go#L352 to avoid one more looping.

I had it here but decided to do it outside to be more cleaner, since we have captured all the IPs now for a given path. I would like to keep it outside.

> You would need to use like https://github.com/minio/minio/blob/master/cmd/net.go#L296

This is not correct this will throw error even for good case when single export is exported over one local IP.